timesketch/lib/analyzers/interface.py

```diff
@@ -219,11 +220,13 @@ def add_view(self, name, query_string=None, query_dsl=None,
         if not query_filter:
             query_filter = {'indices': '_all'}
 
-        view = View(name=name, sketch=self.sql_sketch, user=None,
-                    query_string=query_string, query_filter=query_filter,
-                    query_dsl=query_dsl, searchtemplate=None)
-
+        name = '[{0:s}] {1:s}'.format(analyzer_name, view_name)
+        view = View.get_or_create(name=name, sketch=self.sql_sketch, user=None)
+        view.query_string = query_string
         view.query_filter = view.validate_filter(query_filter)
+        view.query_dsl = query_dsl
+        view.searchtemplate = None
+
         db_session.add(view)
```

Comment:
since you are using the get_or_create function of the BaseModel, which does the db_session.add/db_session.commit if there is no instance, is thsi needed?

Reply:
Yeah, we still need it because the DB object (view) is modified after the get_or_create call. so we need to commit it to the DB with the updated data.